worlds/oot_soh/location_access/overworld/sacred_forest_meadow.py

```diff
@@ -7,6 +7,89 @@
     from worlds.oot_soh import SohWorld
 
 
+class EventLocations(str, Enum):
+    SFM_GOSSIP_STONE = "SFM Gossip Stone"
+    SFM_FAIRY_FOUNTAIN = "SFM Fairy Fountain"
+
+
 def set_region_rules(world: "SohWorld") -> None:
-    player = world.player
+    # SFM Entryway
+    # Connections
+    connect_regions(Regions.SFM_ENTRYWAY, world, [
+        (Regions.LW_BEYOND_MIDO, lambda bundle: True),
+        (Regions.SACRED_FOREST_MEADOW, lambda bundle: is_adult(bundle) or can_kill_enemy(bundle, Enemies.WOLFOS)),
+        (Regions.SFM_WOLFOS_GROTTO, lambda bundle: can_open_bomb_grotto(bundle)),
+    ])
+
+    # Sacred Forest Meadow
+    # Events
+    add_events(Regions.SACRED_FOREST_MEADOW, world, [
+        (EventLocations.SFM_GOSSIP_STONE, Events.CAN_ACCESS_FAIRIES, lambda bundle: call_gossip_fairy_except_suns(bundle)),
+    ])
+    # Location
+    add_locations(Regions.SACRED_FOREST_MEADOW, world, [
+        (Locations.SONG_FROM_SARIA, lambda bundle: is_child(bundle) and has_item(Items.ZELDAS_LETTER, bundle)),
+        (Locations.SHEIK_IN_FOREST, lambda bundle: is_adult(bundle)),
+        (Locations.SFM_GS, lambda bundle: is_adult(bundle) and hookshot_or_boomerang(bundle) and can_get_nighttime_gs(bundle)),
+        (Locations.SFM_MAZE_LOWER_GOSSIP_STONE_FAIRY, lambda bundle: call_gossip_fairy_except_suns(bundle)),
+        (Locations.SFM_MAZE_LOWER_GOSSIP_STONE_BIG_FAIRY, lambda bundle: can_play_song(Items.SONG_OF_STORMS, bundle)),
+        (Locations.SFM_MAZE_UPPER_GOSSIP_STONE_FAIRY, lambda bundle: call_gossip_fairy_except_suns(bundle)),
+        (Locations.SFM_MAZE_UPPER_GOSSIP_STONE_BIG_FAIRY, lambda bundle: can_play_song(Items.SONG_OF_STORMS, bundle)),
+        (Locations.SFM_SARIA_GOSSIP_STONE_FAIRY, lambda bundle: call_gossip_fairy_except_suns(bundle)),
+        (Locations.SFM_SARIA_GOSSIP_STONE_BIG_FAIRY, lambda bundle: can_play_song(Items.SONG_OF_STORMS, bundle)),
+        (Locations.SFM_MAZE_LOWER_GOSSIP_STONE, lambda bundle: True),
+        (Locations.SFM_MAZE_UPPER_GOSSIP_STONE, lambda bundle: True),
+        (Locations.SFM_SARIA_GOSSIP_STONE, lambda bundle: True),
```

Comment:
These ones just hold hints in Ship, they're not needed for us. They can be removed.